### ISSUE TITLE
Provide more context to error messages (logs)

### ISSUE THREAD

**Issue by Swahjak** [mollie/magento2#135]
Currently if (and when) API calls fail the errors wil be logged, but there is little or no context in the error messages. This makes it hard to debug possible issues. Could the order increments be added to the messages to make it easier to find specific issues? Something like `Mollie: Error executing API call (400: Bad Request): The order (10001234) cannot be canceled from state: paid. Documentation: https://docs.mollie.com/guides/handling-errors`

**Comment by Marvin-Magmodules** [comment]
Hi @Swahjak, Thank you for your report. Unfortunatly we don't like to edit existing code that is working properly. So for now we aren't going the change the error messages. We added a [definition of done](https://github.com/mollie/magento2/pull/147/files) to our workflow which contains a check to add context to error messages. Overtime this will improve the thrown error messages.

**Comment by Swahjak** [comment]
@Marvin-Magmodules This is not really a code change, it's just an addition to an existing log message providing context. Currently it's just guessing by timestamp on which order it could possibly be. This isn't a breaking change, just tiny improvement.

**Comment by Frank-Magmodules** [comment]
Hi @Swahjak, You can always open up a pull request to propose those changes, but for now this is the best we can offer.

**Comment by Swahjak** [comment]
I just encountered a nice example. Given the error below, how am I supposed to find out which order is failing / requesting an invalid state?
```
Magento\Framework\Exception\LocalizedException: Mollie: Error executing API call (400: Bad Request): The order cannot be canceled from state: paid. Documentation: https://docs.mollie.com/guides/handling-errors in /app/vendor/mollie/magento2/Model/Client/Orders.php:460
Stack trace:
#0 /app/vendor/mollie/magento2/Model/Mollie.php(374): Mollie\Payment\Model\Client\Orders->cancelOrder(Object(Toppy\Rma\Model\Order\Interceptor))
#1 /app/vendor/mollie/magento2/Observer/OrderCancelAfter.php(56): Mollie\Payment\Model\Mollie->cancelOrder(Object(Toppy\Rma\Model\Order\Interceptor))
#2 /app/vendor/magento/framework/Event/Invoker/InvokerDefault.php(72): Mollie\Payment\Observer\OrderCancelAfter->execute(Object(Magento\Framework\Event\Observer))
#3 /app/vendor/magento/framework/Event/Invoker/InvokerDefault.php(60): Magento\Framework\Event\Invoker\InvokerDefault->_callObserverMethod(Object(Mollie\Payment\Observer\OrderCancelAfter), Object(Magento\Framework\Event\Observer))
#4 /app/vendor/magento/framework/Event/Manager.php(66): Magento\Framework\Event\Invoker\InvokerDefault->dispatch(Array, Object(Magento\Framework\Event\Observer))
#5 /app/generated/code/Magento/Framework/Event/Manager/Proxy.php(95): Magento\Framework\Event\Manager->dispatch('order_cancel_af...', Array)
#6 /app/vendor/magento/module-sales/Model/Order.php(1143): Magento\Framework\Event\Manager\Proxy->dispatch('order_cancel_af...', Array)
#7 [internal function]: Magento\Sales\Model\Order->cancel()
#8 /app/vendor/magento/framework/Data/Collection.php(512): call_user_func_array(Array, Array)
#9 /app/vendor/magento/module-sales/Model/CronJob/CleanExpiredOrders.php(51): Magento\Framework\Data\Collection->walk('cancel')
#10 [internal function]: Magento\Sales\Model\CronJob\CleanExpiredOrders->execute(Object(Magento\Cron\Model\Schedule\Interceptor))
#11 /app/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(317): call_user_func_array(Array, Array)
#12 /app/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(737): Magento\Cron\Observer\ProcessCronQueueObserver->_runJob(1559912400, 1559912402, Array, Object(Magento\Cron\Model\Schedule\Interceptor), 'default')
#13 /app/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(243): Magento\Cron\Observer\ProcessCronQueueObserver->processPendingJobs('default', Array, 1559912402)
#14 /app/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(272): Magento\Cron\Observer\ProcessCronQueueObserver->Magento\Cron\Observer\{closure}('default')
#15 /app/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(244): Magento\Cron\Observer\ProcessCronQueueObserver->lockGroup('default', Object(Closure))
#16 /app/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Cron\Observer\ProcessCronQueueObserver->execute(Object(Magento\Framework\Event\Observer))
#17 /app/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Cron\Observer\ProcessCronQueueObserver\Interceptor->___callParent('execute', Array)
#18 /app/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Cron\Observer\ProcessCronQueueObserver\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\Event\Observer))
#19 /app/generated/code/Magento/Cron/Observer/ProcessCronQueueObserver/Interceptor.php(26): Magento\Cron\Observer\ProcessCronQueueObserver\Interceptor->___callPlugins('execute', Array, Array)
#20 /app/vendor/magento/framework/Event/Invoker/InvokerDefault.php(72): Magento\Cron\Observer\ProcessCronQueueObserver\Interceptor->execute(Object(Magento\Framework\Event\Observer))
#21 /app/vendor/magento/framework/Event/Invoker/InvokerDefault.php(60): Magento\Framework\Event\Invoker\InvokerDefault->_callObserverMethod(Object(Magento\Cron\Observer\ProcessCronQueueObserver\Interceptor), Object(Magento\Framework\Event\Observer))
#22 /app/vendor/magento/framework/Event/Manager.php(66): Magento\Framework\Event\Invoker\InvokerDefault->dispatch(Array, Object(Magento\Framework\Event\Observer))
#23 /app/generated/code/Magento/Framework/Event/Manager/Proxy.php(95): Magento\Framework\Event\Manager->dispatch('default', Array)
#24 /app/vendor/magento/framework/App/Cron.php(86): Magento\Framework\Event\Manager\Proxy->dispatch('default')
#25 /app/vendor/magento/module-cron/Console/Command/CronCommand.php(117): Magento\Framework\App\Cron->launch()
#26 /app/vendor/symfony/console/Command/Command.php(255): Magento\Cron\Console\Command\CronCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#27 /app/vendor/symfony/console/Application.php(893): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#28 /app/vendor/symfony/console/Application.php(262): Symfony\Component\Console\Application->doRunCommand(Object(Magento\Cron\Console\Command\CronCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#29 /app/vendor/magento/framework/Console/Cli.php(102): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#30 /app/vendor/symfony/console/Application.php(145): Magento\Framework\Console\Cli->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#31 /app/bin/magento(23): Symfony\Component\Console\Application->run()
#32 {main}
```

**Comment by Marvin-Magmodules** [comment]
Hi @Swahjak yes this a good example. We have added the entity id of the order number to the log when it is not possible to cancel an order, see [PR 156](https://github.com/mollie/magento2/pull/156).

Also as mention earlier, we  have added a definition of done to our workflow which contains a check to add context to error messages. Overtime this will improve the thrown error messages, see [PR 147](https://github.com/mollie/magento2/pull/147).

**Comment by Frank-Magmodules** [comment]
Added Order ID of the order number to the log when it is not possible to cancel an order and merged this into [v1.6.2](https://github.com/mollie/magento2/releases/tag/v1.6.2) release. Closing this issue now as we have added this to a definition of done to our workflow.
